nightfall-deployer/circuits/common/generic_circuit/Verifiers/verify_commitments.zok

```diff
@@ -0,0 +1,34 @@
+from "hashes/poseidon/poseidon.zok" import main as poseidon
+from "../../utils/structures.zok" import Point
+
+/*
+* Verify that all commitments are correct
+*/
+def main<NumCommitments>(\
+    field packedErcAddress,\
+    field idRemainder,\
+    field[NumCommitments] commitmentHashes,\
+    private field[NumCommitments] newCommitmentValues,\
+    private field[NumCommitments] newCommitmentSalts,\
+    private Point[NumCommitments] recipientPublicKey\
+) -> bool:
+
+    //Check that all the commitments are valid. If NumCommitments equals zero this loop will be ignored
+    for u32 i in 0..NumCommitments do
+        
+        //Calculate the commmitment hash from the newCommitment parameters
+        field commitment = if newCommitmentValues[i] == 0 then  0 else \
```

Comment:
ERC721 tokens have always value = 0. Therefore, the poseidon hash will never be calculated and the verification will fail. I think we also need to check if the commitmentHash is equal to zero. If ERC20 or ERC1155 and value is zero, since the minimum value for this tokens is 1, this will still hold. The equation will also hold for ERC721 since if the commitment is void the hash will be zero.
```suggestion
        field commitment = if newCommitmentValues[i] == 0 && commitmentHashes[i] == 0 then  0 else \
```